frappe/public/js/frappe/form/controls/datetime.js

```diff
@@ -22,5 +25,37 @@ frappe.ui.form.ControlDatetime = frappe.ui.form.ControlDate.extend({
 			}
 		}
 		this._super();
-	}
+	},
+	set_datepicker: function() {
+		this._super();
+		if (this.datepicker.opts.timeFormat.indexOf('s') == -1) {
+			// No seconds in time format
+			this.noSeconds = true;
+			const $tp = this.datepicker.timepicker;
+			$tp.$seconds.parent().css('display', 'none');
+			$tp.$secondsText.css('display', 'none');
+			$tp.$secondsText.prev().css('display', 'none');
+		} else {
+			this.noSeconds = false;
+		}
+	},
+	update_datepicker_position: function() {
+		if (!this.frm) return;
+		// show datepicker above or below the input
+		// based on scroll position
+		const scroll_limit = $(window).scrollTop() + $(window).height();
+		const picker_bottom = this.$input.offset().top + this.$input.outerHeight() + 12 + this.datepicker.$datepicker.outerHeight();
+
+		var position = 'bottom left';
+		//if (picker_top > scroll_limit + window_scroll) {
+		if (picker_bottom >= scroll_limit) {
+			position = 'top left';
+			// bodge around the picker being 30 pixels shorter than
+			// it should be due to hiding seconds
+			if (this.noSeconds) this.datepicker.opts['offset'] = -28;
+		} else {
+			if (this.noSeconds) this.datepicker.opts['offset'] = 12;
```

Comment:
Do these values work properly in all screen sizes? Mobile?

Reply:
This will need a bit more checking (it's been a while since I last tried it and it was a different ERPNext version).

Reply:
This has been rejigged a bit. There is now only one 'magic' number to work around mispositioning of the timepicker when opening upwards (it was out by 42px or so). All other numbers (such as the position of the bottom of the header) are taken from the elements. I have tested this in the Chrome devtools 'mobile' simulator thing and it looks like it all works and there is no significant difference.